src/v/cluster/rm_stm.cc

```diff
@@ -187,6 +187,31 @@ struct tx_snapshot_v0 {
     std::vector<seq_entry_v0> seqs;
 };
 
+struct seq_cache_entry_v1 {
```

Comment:
> rm_stm: prepare to use kafka offset based cache

not really sure what that means. the changes in this commit are related to changes in serialized data formats and the issues surrounding that change and backwards compatibility need to be explained.

Reply:
This commit doesn't change anything substantial, I move `seq_cache_entry` into `seq_cache_entry_v1` introduce new version of the snapshot but if you check `seq_cache_entry` it's exactly the same as `seq_cache_entry_v1`; the next commit changes `seq_cache_entry` to use new type.

Changing all at once create too much noise so this commit is to prepare to use kafka offset and the next commit actually starts using it. I separated them to control the attention of a reviewer.

Reply:
Updated the commit message